src/comments/comment.entity.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  OneToMany,
+  OneToOne,
+  JoinColumn,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+  DeleteDateColumn,
+} from 'typeorm';
+import { Answer } from '../answer/answer.entity';
+import { User } from '../users/users.entity';
+import { UserDto } from '../users/DTO/user.dto';
+import { QuestionDto } from '../questions/DTO/question.dto';
+import { Question } from '../questions/questions.entity';
+
+@Entity()
+export class Comment {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  content: string;
+
+  @Column()
+  commentableId: number;
+
+  @Column()
+  commentableType: string;
+
+  @Column()
+  agreecount: number;
+
+  @Column()
+  disagreecount: number;
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: 'user_id' })
+  user: UserDto;
+
+  @OneToMany(() => Comment, (comment) => comment.parentComment)
```

Comment:
No `CommentMemberShip` or `CommentAnswerShip`. use polymorphic relations, like
```typescript
CommentableRelationship
{
  commentable_type: CommentableType,
  commentable_id: number,
  comment_id: number
}
```
And do not include parentComment or subComments in columns of Comment table. If querying comments of a commetable is needed, query in CommentableRelationship table.